### ISSUE TITLE
Rescoring drops query weight / rescore weight when rescore query retwritten

### ISSUE THREAD

**Issue by softwaredoug** [elastic/elasticsearch#27979]
**Describe the feature**:

*Elasticsearch version** (`bin/elasticsearch --version`):

6.1.1

**Plugins installed**: [LTR]

**JVM version** (`java -version`):

```
(venv) doug@wiz$~/ws/elasticsearch-learning-to-rank(es6) $ java -version
java version "1.8.0_121"
Java(TM) SE Runtime Environment (build 1.8.0_121-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.121-b13, mixed mode)
```

**OS version** (`uname -a` if on a Unix-like system):

```
Darwin wiz.local 17.3.0 Darwin Kernel Version 17.3.0: Thu Nov  9 18:09:22 PST 2017; root:xnu-4570.31.3~1/RELEASE_X86_64 x86_64
```

**Description of the problem including expected versus actual behavior**:

Rescore query weights are not carried through when the rescore query is rewritten. Debugging shows that weights appear to be dropped when [query rescore builder is rewritten](https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/search/rescore/QueryRescorerBuilder.java#L244). 

You can see in the code: 

```
    @Override
    public QueryRescorerBuilder rewrite(QueryRewriteContext ctx) throws IOException {
        QueryBuilder rewrite = queryBuilder.rewrite(ctx);
        if (rewrite == queryBuilder) {
            return this;
        }
        return new QueryRescorerBuilder(rewrite);
    }
```

That various rescore settings are not carried through to the rewritten QueryRescoreBuilder. Only the rewritten query builder is carried through, not the weights that are also part of this. IE I would expect something like:

```
    @Override
    public QueryRescorerBuilder rewrite(QueryRewriteContext ctx) throws IOException {
        QueryBuilder rewrite = queryBuilder.rewrite(ctx);
        if (rewrite == queryBuilder) {
            return this;
        }
        return new QueryRescorerBuilder(rewrite)
                           .setRescoreQueryWeight(rescoreQueryWeight) 
                           . (additional settings carried through )
    }
```

**Steps to reproduce**:

See this [integration test](https://github.com/o19s/elasticsearch-learning-to-rank/blob/es6_1/src/test/java/com/o19s/es/ltr/query/StoredLtrQueryIT.java#L58)

**Provide logs (if relevant)**:

**Comment by softwaredoug** [comment]
Do not know the protocol, but should this be closed with the merge of #27981 ?

**Comment by jimczi** [comment]
I forgot to add the magic words in the commit message. Thanks for checking.
